src/synology_dsm/synology_dsm.py

```diff
@@ -110,9 +110,9 @@ def __init__(
 
         # Build variables
         if use_https:
-            self._base_url = f"https://{dsm_ip}:{dsm_port}"
+            self._base_url = URL(f"https://{dsm_ip}:{dsm_port}")

```

Comment:
By building this here we avoid calling `[_make_netloc](https://github.com/aio-libs/yarl/blob/bd5ff24f8a5a3347bf3251237b6d32c40e21b928/yarl/_url.py#L197)` on every request which has to make an ip_address object and than convert it back to a string